logfire/variables/remote.py

```
@@ -314,20 +322,43 @@ def _worker(self):
 
+            # Gap 5: after an SSE-event-triggered forced refresh, schedule a single follow-up
+            # refresh ~2s later.  The server-side in-memory cache (platform PR #28877) has a 1s
+            # per-pod TTL, so the immediate refetch can race and return the pre-update config.
+            # One debounced follow-up guarantees convergence without adding new threads.
+            if force:
+                self._followup_refresh_at = time.monotonic() + 2.0
+
             # Gap 3: add uniform +/-10% jitter to the wait so that fleets deployed
             # simultaneously don't poll in lockstep.
             base_interval = self._polling_interval.total_seconds()
             wait_timeout = base_interval * random.uniform(0.9, 1.1)
```

Ruff (0.15.21)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)